### ISSUE TITLE
BPF filter doesn't sniff ARP over Dot1Q

### ISSUE THREAD

**Issue by kandaur** [secdev/scapy#1734]
#### Brief description

Sniff function with ARP BPF filter applied (sniff(iface='lo', filter='arp')) can't handle ARP over Dot1Q frames.

#### Environment

-   Scapy version: 2.4.0
-   Python version: 2.7.15rc1 (default, Nov 12 2018, 14:31:15), 3.6.7 (default, Oct 22 2018, 11:32:17)
-   Operating System: Ubuntu 18.04.1 LTS

#### How to reproduce

1) In 1st scapy process Start sniffing process with ARP filter: 

`
sniff(iface='lo', filter='arp')
`

2) In 2nd scapy process Start sending the following frame:
```
arp_over_dot1q.show()
###[ Ethernet ]###
  dst= 00:00:33:44:11:aa
  src= 00:00:00:00:00:11
  type= n_802_1Q
###[ 802.1Q ]###
     prio= 0
     id= 0
     vlan= 80
     type= ARP
###[ ARP ]###
        hwtype= 0x1
        ptype= IPv4
        hwlen= 6
        plen= 4
        op= who-has
        hwsrc= 00:00:00:00:00:00
        psrc= 10.1.1.1
        hwdst= 00:00:00:00:00:00
        pdst= 10.1.1.2
###[ Padding ]###
           load= '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'

arp_over_dot1q.command()
"Ether(src='00:00:00:00:00:11', dst='00:00:33:44:11:aa', type=33024)/Dot1Q(vlan=80, type=2054, id=0, prio=0)/ARP(hwdst='00:00:00:00:00:00', ptype=2048, hwtype=1, psrc='10.1.1.1', hwlen=6, plen=4, pdst='10.1.1.2', hwsrc='00:00:00:00:00:00', op=1)/Padding(load='\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00\\x00')"

sendp(arp_over_dot1q * 10, iface='lo')
```
3) Stop sniffing process and verify captured traffic

#### Actual result

No any traffic captured:
`
^C<Sniffed: TCP:0 UDP:0 ICMP:0 Other:0>
`
#### Expected result

10 frames captured:
`
^C<Sniffed: TCP:0 UDP:0 ICMP:0 Other:10>
`
#### Related resources

1) When no any filter is applied with sniffer, all frames are successfully captured
2) Traffic can be successfully sniffed by wireshark with 'arp' capture filter (not display filter):
![selection_039](https://user-images.githubusercontent.com/36377004/49642574-1dc0c100-fa1c-11e8-9542-5ed940e9fc7c.png)

**Comment by p-l-** [comment]
Hi,

The loopback interface is not a "regular" interface. Can you reproduce this with a regular, ethernet device?

Does the filter work properly when used with `tcpdump`?

**Comment by kandaur** [comment]
I just successfully reproduced the issue on regular 1G copper interface.
Also I reproduced it when tried to sniff pcap file in offline mode:
```
arp_over_dot1q = rdpcap('/home/okondaur/Downloads/sniff_file.pcap')
>>> arp_over_dot1q
<sniff_file.pcap: TCP:0 UDP:0 ICMP:0 Other:10>

>>> print(arp_over_dot1q[0].show())
###[ Ethernet ]###
  dst= 00:00:33:44:11:aa
  src= 00:00:00:00:00:11
  type= n_802_1Q
###[ 802.1Q ]###
     prio= 0
     id= 0
     vlan= 80
     type= ARP
###[ ARP ]###
        hwtype= 0x1
        ptype= 0x800
        hwlen= 6
        plen= 4
        op= who-has
        hwsrc= 00:00:00:00:00:00
        psrc= 10.1.1.1
        hwdst= 00:00:00:00:00:00
        pdst= 10.1.1.2
###[ Padding ]###
           load= '\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'

>>> for t in arp_over_dot1q[:1]:
...   sniff(offline=t, filter='arp')
...
reading from file -, link-type EN10MB (Ethernet)
<Sniffed: TCP:0 UDP:0 ICMP:0 Other:0>

>>> for t in arp_over_dot1q[:1]:
...   sniff(offline=t, filter='vlan 80')
...
reading from file -, link-type EN10MB (Ethernet)
<Sniffed: TCP:0 UDP:0 ICMP:0 Other:1>
```

Filter works fine on `tcpdump`. Please see screenshot attached at my previous message. It was captured with `arp` specified as [capture filter  ](https://wiki.wireshark.org/CaptureFilters)

**Comment by p-l-** [comment]
Can you please sniff with Scapy on the interface, and, **at the same time**, sniff with tcpdump, with the exact same filter?

**Comment by kandaur** [comment]
1) For egress traffic (on the same interface from which traffic is generated) no traffic was captured by both scapy and tcpdump with filter:

```
sendp(arp_over_dot1q, iface='eth1')
--------------------------
sniff(iface='eth1', filter='arp')
tcpdump -ni eth1 -vvv arp
```

2) without filter the captured output is different for egress traffic:

- also nothing for scapy:
```
sendp(arp_over_dot1q, iface='eth1')
--------------------------
sniff(iface='eth1')
```

- but tcpdump captured frames with dot1q tag

```
sendp(arp_over_dot1q[:2], iface='eth1')
--------------------------
tcpdump -ni eth1 -vvv arp -w /tmp/111.pcap
```
see 111.pcap in [tcpdump.zip](https://github.com/secdev/scapy/files/2666884/tcpdump.zip)

3) For ingress traffic (traffic was generated to eth1 interface, connected to switch port in trunk mode, and received by eth2, connected to another switch port in trunk mode) the captured output is different:

-  scapy captured frames witout dot1q tag

```
sendp(arp_over_dot1q[:2], iface='eth1')
--------------------------
>>> tt = sniff(iface='eth2', filter='arp and ether host 00:00:00:00:00:11')
^C>>> tt
<Sniffed: TCP:0 UDP:0 ICMP:0 Other:2>
>>> tt[0]
<Ether  dst=00:00:33:44:11:aa src=00:00:00:00:00:11 type=ARP |<ARP  hwtype=0x1 ptype=0x800 hwlen=6 plen=4 op=who-has hwsrc=00:00:00:00:00:00 psrc=10.1.1.1 hwdst=00:00:00:00:00:00 pdst=10.1.1.2 |<Padding  load='\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' |>>>
>>> tt[1]
<Ether  dst=00:00:33:44:11:aa src=00:00:00:00:00:11 type=ARP |<ARP  hwtype=0x1 ptype=0x800 hwlen=6 plen=4 op=who-has hwsrc=00:00:00:00:00:00 psrc=10.1.1.1 hwdst=00:00:00:00:00:00 pdst=10.1.1.2 |<Padding  load='\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00' |>>>
```

- tcpdump captured frames with dot1q tag

```
sendp(arp_over_dot1q[:2], iface='eth1')
--------------------------
tcpdump -ni eth2 -vvv arp and ether host 00:00:00:00:00:11 -w /tmp/123.pcap
```
see 123.pcap in [tcpdump.zip](https://github.com/secdev/scapy/files/2666884/tcpdump.zip)

**Comment by p-l-** [comment]
OK I think this issue is a duplicate (even if you bring a new symptom, I think it is the same bug) of #969. What do you think?

**Comment by p-l-** [comment]
OK, closed as duplicate of #969.
